src/dbg/module.cpp

```diff
@@ -400,4 +414,30 @@ bool ModAddImportToModule(duint Base, const MODIMPORTINFO & importInfo)
     pImports->push_back(importInfo);
 
     return true;
-}
\ No newline at end of file
+}
+
+int ModGetParty(duint Address)
+{
+    SHARED_ACQUIRE(LockModules);
+
+    auto module = ModInfoFromAddr(Address);
+
+    // If the module is not found, it is an user module
+    if (!module)
+        return 0;
+
+    return module->party;
+}
+
+void ModSetParty(duint Address, int Party)
+{
+    SHARED_ACQUIRE(LockModules);

```

Comment:
Why not use InterlockedExchange then? Or at least write a comment explaining it, since it might raise issues later.